File: app/src/main/java/com/example/smarthome/user/viewModels/UserViewModel.java
package com.example.smarthome.user.viewModels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

//import com.example.smarthome.user.models.User;
import com.example.smarthome.user.models.LoginBody;
import com.example.smarthome.user.models.User;
import com.example.smarthome.user.services.UserService;

public class UserViewModel extends ViewModel {

    private UserService service = UserService.getInstance();

    public MutableLiveData<String> getSignupMsg(){
        return service.getSignupMsg();
    }
    public MutableLiveData<String> getLoginMsg(){
        return service.getLoginMsg();
    }
    public LiveData<Boolean> getIsSignedIn(){return service.getSignedIn();}
    public LiveData<Boolean> showProgressBar(){return service.getShowProgressBar();}


    public void signin(LoginBody user){
        service.signin(user);
    }

    public void signup(User user){
        service.signup(user);
    }
}

File: app/src/main/java/com/example/smarthome/user/viewModels/UserViewModel.kt
package com.example.smarthome.user.viewModels

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.smarthome.user.models.LoginRequest
import com.example.smarthome.user.models.User
import com.example.smarthome.user.services.UserService
import com.example.smarthome.utilities.Resource

class UserViewModel : ViewModel(){

    private val service = UserService.getInstance()

    fun getIsSignedIn(): LiveData<Boolean> = service.signedIn
    fun getServerStatusLD(): LiveData<Boolean> = service.serverStatusLD
    fun getStatus(): LiveData<Resource<User>> = service.status


    fun signin(loginRequest: LoginRequest){
        service.signin(loginRequest)
    }

    fun signup(user: User){
        service.signup(user)
    }

    fun signOut(){
        service.signOut()
    }

    fun getServerStatus(){
        service.getServerStatus()
    }
}